src/asio_service.cxx

```diff
@@ -610,7 +610,7 @@ class rpc_session : public std::enable_shared_from_this<rpc_session> {
             const uint16_t CUR_HINT_VERSION = 0;
             bs.put_u16(CUR_HINT_VERSION);
             bs.put_u16(sizeof(ulong));
-            bs.put_u64(resp->get_next_batch_size_hint_in_bytes());
+            bs.put_i64(resp->get_next_batch_size_hint_in_bytes());
```

Comment:
During a rolling upgrade, old Raft instance with `u64` and new Raft instance with `i64` will be running in the same group. Will it be safe? If `i64` server returns a negative number, it is interpreted as a big number for `u64` server, and the only impact I can imagine is that there is no throttling, right?

Reply:
Yes. I think the only impact is that `hint size` will not take effect. Besides, in monstordb, we still have 16M restriction. So the whole throttling should still be there.